include/irr/asset/filters/CSwizzleAndConvertImageFilter.h

```diff
@@ -126,22 +130,31 @@ class CSwizzleAndConvertImageFilter : public CImageFilter<CSwizzleAndConvertImag
 			if (state->inImage->getCreationParameters().format!=inFormat)
 				return false;
 
-			if (state->inImage->getCreationParameters().format!=outFormat)
+			if (state->outImage->getCreationParameters().format!=outFormat)
 				return false;
 
 			return true;
 		}
 
 		static inline bool execute(state_type* state)
 		{
+			if (!validate(state))
+				return false;
+
 			const auto blockDims = asset::getBlockDimensions(inFormat);
 			#ifdef _IRR_DEBUG
 				assert(blockDims.z==1u);
 				assert(blockDims.w==1u);
 			#endif
+
+			typedef std::conditional<asset::isIntegerFormat<inFormat>(), uint64_t, double>::type decodeBufferType;
+			typedef std::conditional<asset::isIntegerFormat<outFormat>(), uint64_t, double>::type encodeBufferType;
+
 			auto perOutputRegion = [&blockDims,&state](const CMatchedSizeInOutImageFilterCommon::CommonExecuteData& commonExecuteData, CBasicImageFilterCommon::clip_region_functor_t& clip) -> bool
 			{
-				auto swizzle = [&commonExecuteData,&blockDims,&state](uint32_t readBlockArrayOffset, core::vectorSIMDu32 readBlockPos)
+				const auto clampBufferChannelsAmount = asset::getFormatChannelCount(outFormat);
```

Comment:
is there a compile time version ? like `asset:getFormatChannelCount<E_FORMAT format>(void)` ?

Reply:
?

Reply:
oh sorry, no there isn't

Reply:
make one.

Reply:
copy!

Reply:
added and in use!